src/main/java/org/embulk/output/mailchimp/helper/MailChimpRetryable.java

```diff
@@ -0,0 +1,224 @@
+    private static TokenHolder tokenHolder;
+    protected StringJsonParser jsonParser = new StringJsonParser();
+    private String authorizationHeader;
+
+    public MailChimpRetryable(final PluginTask pluginTask)
+    {
+        this.retryHelper = new Jetty92RetryHelper(pluginTask.getMaximumRetries(),
+                                                  pluginTask.getInitialRetryIntervalMillis(),
+                                                  pluginTask.getMaximumRetryIntervalMillis(),
+                                                  new DefaultJetty92ClientCreator(pluginTask.getTimeoutMillis(),
+                                                                                  pluginTask.getTimeoutMillis()));
+        this.pluginTask = pluginTask;
+        authorizationHeader = buildAuthorizationHeader(pluginTask);
+    }
+
+    public String get(final String path)
+    {
+        return sendRequest(path, null);
+    }
+
+    public String post(final String path, String contentType, String body)
+    {
+        return sendRequest(path, new StringContentProvider(contentType, body, Charsets.UTF_8));
+    }
+
+    private String sendRequest(final String path, final StringContentProvider contentProvider)
+    {
+        return retryHelper.requestWithRetry(
+                new StringJetty92ResponseEntityReader(READER_TIMEOUT_MILLIS),
+                new Jetty92SingleRequester()
+                {
+                    @Override
+                    public void requestOnce(HttpClient client, Response.Listener responseListener)
+                    {
+                        createTokenHolder(client);
+                        Request request = client.newRequest(tokenHolder.getEndpoint() + path)
+                                .header(AUTHORIZATION, authorizationHeader)
+                                .method(GET);
+                        if (contentProvider != null) {
+                            request = request.method(POST).content(contentProvider);
+                        }
+                        request.send(responseListener);
+                    }
+
+                    @Override
+                    protected boolean isResponseStatusToRetry(Response response)
+                    {
+                        // Retry if it's a server or rate limit exceeded error
+                        return (response.getStatus() != 500 && response.getStatus() / 100 != 4) || response.getStatus() == 429;
+                    }
+
+                    @Override
+                    protected boolean isExceptionToRetry(Exception exception)
+                    {
+                        // This check is to make sure if the original exception is retryable, i.e.
+                        // server not found, internal server error...
+                        if (exception instanceof ConfigException || exception instanceof ExecutionException) {
+                            return toRetry((Exception) exception.getCause());
+                        }
```

Comment:
Does this block might duplicated with `return (response.getStatus() != 500 && response.getStatus() / 100 != 4) || response.getStatus() == 429;` ?

In my understanding, checking status code?

Reply:
There are some errors is not `HttpResponseException`, we will call `super.toRetry`